src/Model/Contact.php

```diff
@@ -2184,7 +2190,7 @@ private static function updateFromProbeArray(int $id, array $ret)
 		}
 
 		$update = false;
-		$guid = $ret['guid'] ?? '';
+		$guid = ($ret['guid'] ?? '') ?: Item::guidFromUri($ret['url'], parse_url($ret['url'], PHP_URL_HOST));
```

Comment:
The `$string ?? '' ?: $else` sequence (or `$number ?? 0 ?: $else` or `$bool ?? false ?: $else`) is the equivalent of `!empty($var) ? $var : $else`, no need for parentheses in my mind.

```suggestion
		$guid = $ret['guid'] ?? '' ?: Item::guidFromUri($ret['url'], parse_url($ret['url'], PHP_URL_HOST));
```

Reply:
The parentheses help me in understanding what's going on there.

Reply:
Null-coalescing `??` and Falsy-coalescing operators `?:` can be freely chained, think of them as arrows meaning "or else fall back to"

In this case it would look like this: `$ret['guid'] → '' → Item::guidFromUri(...)` and you can see parentheses are superfluous.

Reply:
It's just like punctuation: There are places where there are superfluous - but they improve the reading.